src/openloco/windows/construction/ConstructionTab.cpp

```diff
@@ -396,14 +396,10 @@ namespace openloco::ui::windows::construction::construction
     // 0x0049DAF3
     static void disableTrackSlopes(window* self, track_object trackObj, uint64_t disabledWidgets)
     {
-        auto trackPieces = trackObj.track_pieces & ((1 << 5) | (1 << 8));
-
-        if (trackPieces != ((1 << 5) | (1 << 8)))
+        if (trackObj.track_pieces != ((1 << 5) | (1 << 8)))
```

Comment:
I'm not sure this is correct. `trackObj.trackPieces` could have other bits set as well as 5 and 8 and still disable the slope widgets
```suggestion
        if (!(trackObj.track_pieces & (1 << 5)) && !(trackObj.track_pieces &  (1 << 8)))
```

Reply:
The local variable `trackPieces` was defined as `trackObj.track_pieces & ((1 << 5) | (1 << 8))` though. That throws away any information other than the `((1 << 5) | (1 << 8))` mask. That's why I thought we might as well just simplify things.

Reply:
Bits 5, 6, 7, 8 are most likely the slope it's either gentle, steep, up or down. The old version disabled slopes if either of them are set. You have changed it to disabled if both of them are not set. This is subtly different and will likely disable it in all cases as track pieces would never have both the gentle and steep bits set.